src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UpdateChartFilesService.cs

```diff
@@ -0,0 +1,168 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Chart;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using static GovUk.Education.ExploreEducationStatistics.Common.BlobContainerNames;
+using static GovUk.Education.ExploreEducationStatistics.Common.Services.FileStoragePathUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class UpdateChartFilesService : IUpdateChartFilesService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly ILogger _logger;
+
+        private readonly string _storageConnectionString;
+        
+        public UpdateChartFilesService(
+            IConfiguration configuration,
+            ContentDbContext contentDbContext,
+            ILogger<UpdateChartFilesService> logger)
+        {
+            _storageConnectionString = configuration.GetValue<string>("CoreStorage");
+            _contentDbContext = contentDbContext;
+            _logger = logger;
+        }
+
+        public async Task<Either<ActionResult, bool>> UpdateChartFiles()
+        {
+            var refs = _contentDbContext
+                .ReleaseFileReferences
+                .Where(rfr => rfr.ReleaseFileType == ReleaseFileTypes.Chart)
+                .ToList();
+
+            var result = await UpdateChartFilesAsync(refs);
+
+            if (result.IsLeft)
```

Comment:
No need to do any of this as the success/failure is already represented by the `Either` itself, just do:

```csharp
return await UpdateChartFiles(refs);
```